WordPress/Classes/Services/NullBlogPropertySanitizer.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+
+/// Delete invalid rows in the database whose required blog properties are NULL
+///
+/// This is born from the investigation of the crash described in #12028 (https://git.io/JePdZ).
+/// Unfortunately, we could not find the cause of it. The findings are described in that issue.
+///
+/// This tries to “fix” the issue by deleting the orphaned entities like `Post` whose `blog` were
+/// set to NULL. They are currently inaccessible because they are not attached to any `Blog`. But
+/// leaving them in the database would cause a crash if Core Data tries to save new entities.
+///
+@objc class NullBlogPropertySanitizer: NSObject {
+    static let lastSanitizationVersionNumber = "null-property-sanitization"
+
+    private let store: KeyValueDatabase
+    private let context: NSManagedObjectContext
+
+    @objc init(context: NSManagedObjectContext) {
+        store = UserDefaults.standard
+        self.context = context
+    }
+
+    init(store: KeyValueDatabase, context: NSManagedObjectContext) {
+        self.store = store
+        self.context = context
+    }
+
+    @objc func sanitize() {
+        guard appWasUpdated() else {
+            return
+        }
+
+        self.store.set(self.currentBuildVersion(), forKey: Self.lastSanitizationVersionNumber)
```

Comment:
```suggestion
        store.set(currentBuildVersion(), forKey: Self.lastSanitizationVersionNumber)
```

No need for the extra `self.`

Reply:
Removed in 90f34bfffbcd47ce887c4bd41ea9c505c7d590ca.